src/sas/sascalc/corfunc/corfunc_calculator.py

```diff
@@ -1,290 +1,670 @@
-        :param completefn: The function to call when the transform calculation
-            is complete
-        :param updatefn: The function to call to update the GUI with the status
-            of the transform calculation
-        :return: The transformed data
+        Calculate the Porod parameters
         """
-        if self._transform_thread is not None:
-            if self._transform_thread.isrunning(): return
-
-        if background is None: background = self.background
-
-        if trans_type == 'fourier':
-            self._transform_thread = FourierThread(self._data, extrapolation,
-                                                   background, completefn=completefn,
-                                                   updatefn=updatefn)
-        elif trans_type == 'hilbert':
-            self._transform_thread = HilbertThread(self._data, extrapolation,
-                                                   background, completefn=completefn, updatefn=updatefn)
-        else:
-            err = ("Incorrect transform type supplied, must be 'fourier'",
-                " or 'hilbert'")
-            raise ValueError(err)
 
-        self._transform_thread.queue()
+        if self._data is None:
+            raise ValueError("Data not set")
+
+        if self._extrapolation_parameters is None:
+            raise ValueError("Extrapolation settings not specified")
+
+        if self._porod.allow_fit:
+            # Fit the last section only
+
+            q = self.data.x
+
+            point_2 = self._extrapolation_parameters.point_2
+            point_3 = self._extrapolation_parameters.point_3
+            mask = np.logical_and(q > point_2, q < point_3)
+
+            # Returns an array where the 1st and 2nd elements are the values of k
+            # and sigma for the best-fit Porod function
+            K, sigma, _ = CorfuncCalculator.calculate_porod_parameters(q[mask], self.data.y[mask])
+
+            self._porod.data = PorodData(K=K, sigma=sigma)
 
-    def transform_isrunning(self):
-        if self._transform_thread is None: return False
-        return self._transform_thread.isrunning()
 
-    def stop_transform(self):
-        if self._transform_thread.isrunning():
-            self._transform_thread.stop()
 
-    def extract_parameters(self, transformed_data) -> Optional[ExtractedParameters]:
+    def _calculate_guinier_parameters(self):
+        # Smooths between the best-fit porod function and the data to produce a
+        # better fitting curve
+        # Returns parameters for the best-fit Guinier function
+
+
+        if self._data is None:
+            raise ValueError("Data not set")
+
+        if self._extrapolation_parameters is None:
+            raise ValueError("Extrapolation settings not specified")
+
+        if self._background_subtracted is None:
+            raise ValueError("Backgroundless data not set")
+
+        if self._guinier.allow_fit:
```

Comment:
Yeah, it's quite likely that CorfuncCalculator will be used in user scripts, so the interface is more fleshed out than it would need to be otherwise. This is also the reason for the function at the bottom of `corfunc_calculator.py`.